src/hotspot/share/gc/parallel/psOldGen.cpp

```diff
@@ -173,6 +173,50 @@ HeapWord* PSOldGen::allocate(size_t word_size) {
   return res;
 }
 
+/*
+ * Divide space into blocks, processes block begins at
+ * bottom + block_index  * (IterateBlockSize / HeapWordSize).
+ * NOTE:
+ * - The initial block start address may not be a valid
+ * object address, _start_array is used to correct it.
+ *
+ * - The end address is not necessary to be object address.
+ *
+ * - If there is an object that crosses blocks, it is
+ * processed by the worker that owns the block within
+ * which the object starts.
+ *
+ */
+void PSOldGen::block_iterate(ObjectClosure* cl, uint block_index) {
+  MutableSpace *space = object_space();
```

Comment:
Similar to `object_iterate()` this method should be implemented in the underlying `MutableSpace`, not in `PSOldGen` and just forward to the `MutableSpace` here. While the current design may be debatable, we should adhere to it in new code.

Reply:
Hi @tschatzl, 
  IMHO, the block_iterate() uses PSOldGen's start_array() to identify the object in the block, and without this kind of mechanism, it may not easy to devide Mutablespace into blocks and iterate objects in those blocks, so I think it maybe better to leave it here only for PSOldGen.  what do you think?

Thanks,
Lin